src/main/java/org/opensearch/neuralsearch/processor/chunker/DelimiterChunker.java

```diff
@@ -88,4 +137,24 @@ public List<String> chunk(final String content, final Map<String, Object> runtim
     public String getAlgorithmName() {
         return ALGORITHM_NAME;
     }
+
+    private List<String> splitByLength(final String text, final int maxLength) {
+        List<String> result = new ArrayList<>();
+        int start = 0;
+
+        while (start < text.length()) {
+            int end = Math.min(start + maxLength, text.length());
+
+            // If a delimiter exists within the chunk and doesn't end at the boundary,
+            // adjust to include the full delimiter
+            if (!text.substring(start, end).endsWith(delimiter) && text.indexOf(delimiter, start) < end) {
```

Comment:
text: ab<tags>
delimiter: <tags>
maxLength: 3
Then, the result will be `ab<tags>` which is over maxLength?
If this behavior is correct, we might better change the method name to reflect what it actually does. It does not simply split by length.